### PR TITLE
Create a bash-based fixing plugin

### LINKED ISSUE #9: Add Bash-Script-based fixing plugin
Accepts the following user-inputs:
- bash script
- k/v map as inputs for the script

further inputs are inferred from both the instance model and the issue being handled.

### PR DESCRIPTION
**Pull request by ghareeb-falazi**
Create an issue fixing plugin that is capable of running a user-defined bash script over ssh on a ubuntu-based (virtual-)machine.

**Possible Inputs for the Plugin**
-

_From the Issue properties_
-
The plugin expects the `"CHECKER_COMPONENT_ID"` property. This property points to a component of type `Compute` in the instance model that has the issue.
The referred component is expected to have the following properties:
- `os_family`: with the value `linux`.
- `machine_image`: with the value `ubuntu`.
- `public_address`: with the hostname/ip address of the ubuntu (virtual-)machine.
- (optional) `private_key_path`: with the path (on the server) to the private key that allows connecting to the (virtual-)machine via ssh. If this is property has a value, the plugin configuration entry `default-private-key-path` (see below) is ignored.

_From the Configuration Entries_
-
The plugin has the following configuration entries:
- `script`: the bash script to be executed via ssh.
- `username`: the username to be used when connecting to the ubuntu (virtual-)machine.
- (optional) `default-private-key-path`: the path (on the iacmf server) to the private key that allows to connect to the ubuntu (virtual-)machine. This entry will be used iff the instance model node that has the issue does not define a property `private_key_path` (see above). At least one of these two values must be set.
- (optional) `compliance-rule-arguments`: a comma-separated list of compliance rule parameter names. If this value is set, the plugin will retrieve the referenced attributes and pass their values to the bash script as command-line arguments in the same order specified in this list.

_From the Compliance Rule_
-
If the plugin has a value for the optional configuration entry `compliance-rule-arguments` (see above). Then the corresponding compliance rule attributes will be used as command-line arguments to the script that will be executed on the ubuntu (vritual-)machine via ssh.

Closes #9

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    - `public_address`: with the hostname/ip address of the ubuntu (virtual-)machine.
+    - (optional) `private_key_path`: with the path (on the server) to the private key that allows connecting to the (
+      virtual-)machine via ssh. If this is property has a value, the plugin configuration
+      entry `default-private-key-path` (see below) is ignored.
+
+
+2. _From the Configuration Entries_
+
+   The plugin has the following configuration entries:
+    - `script`: the bash script to be executed via ssh.
```
Comment (nyuuyn) [inline README.md:38]: will this be a file reference later? the complete file as text?

File: src/test/resources/sec/cloud.key [hunk 2]
```diff
@@ -0,0 +1,39 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (nyuuyn) [inline src/test/resources/sec/cloud.key:1]: uff, this is brutal, maybe add this as a secret to the github organisation and then we store the key as a file in github workflow ? Or just pass it as an environment variable to the system?

**Review by nyuuyn — APPROVED** [review]
LGTM, except the key, we should probably fix that :D

File: README.md (hunk 1 above)
Comment (ghareeb-falazi) [reply README.md:38]: only the command to be executed using the `VirtualMachine.execCommand` method.

File: src/test/resources/sec/cloud.key (hunk 2 above)
Comment (ghareeb-falazi) [reply src/test/resources/sec/cloud.key]: You are right.
I removed the key file.
I added the key as a repository secret, and fed it into the maven build stage in the Github workflow as an environment variable called `VM_PRIVATE_KEY`.
The test reads the private key from that environment variable. In order to allow running the tests locally, you can add a base64 version of the private key as an environment variable directly in the test configuration.
I also annotated the test class in a way that disables it if the environment variable is not available.

**Pull request merged**